devito/ir/iet/visitors.py

```diff
@@ -487,25 +513,50 @@ def visit_Operator(self, o):
                                                     self._args_decl(i.root.parameters)))
                 efuncs.extend([self._visit(i.root), blankline])
 
-        # Header files, extra definitions, ...
-        header = [c.Define(*i) for i in o._headers] + [blankline]
-        includes = [c.Include(i, system=(False if i.endswith('.h') else True))
-                    for i in o._includes]
-        includes += [blankline]
-        cdefs = [i._C_typedecl for i in o.parameters if i._C_typedecl is not None]
-        for i in o._func_table.values():
-            if i.local:
-                cdefs.extend([j._C_typedecl for j in i.root.parameters
-                              if j._C_typedecl is not None])
-        cdefs = filter_sorted(cdefs, key=lambda i: i.tpname)
-        if o._compiler.src_ext in ('cpp', 'cu'):
-            cdefs += [c.Extern('C', signature)]
-        cdefs = [i for j in cdefs for i in (j, blankline)]
-
-        return c.Module(header + includes + cdefs +
+        # Definitions
+        headers = [c.Define(*i) for i in o._headers] + [blankline]
+
+        # Header files
+        includes = self._operator_includes(o) + [blankline]
+
+        # Type declarations
+        typedecls = self._operator_typedecls(o, mode)
+        if mode in ('all', 'public') and o._compiler.src_ext in ('cpp', 'cu'):
+            typedecls.append(c.Extern('C', signature))
+        typedecls = [i for j in typedecls for i in (j, blankline)]
+
+        return c.Module(headers + includes + typedecls +
                         esigns + [blankline, kernel] + efuncs)
 
 
+class CInterface(CGen):
+
+    def _operator_includes(self, o):
+        includes = super()._operator_includes(o)
+        includes.append(c.Include("%s.h" % o.name, system=False))
```

Comment:
why this not in `_operator_includes` ?

Reply:
well when we normally execute devito and run `op.apply` there's no Operator-specific header file to include --it's all in the `.c`, no? :)